datashader/resampling.py

```diff
@@ -251,7 +251,9 @@ def _resample_2d(src, mask, use_mask, ds_method, us_method, fill_value, mode_ran
     downsampling_method = DOWNSAMPLING_METHODS[ds_method]
     upsampling_method = UPSAMPLING_METHODS[us_method]
 
-    if out_w < src_w and out_h < src_h:
+    if src_h == 0 or src_w == 0 or out_h == 0 or out_w == 0:
+       return np.zeros((out_h, out_w), dtype=src.dtype)
```

Comment:
Are src_h and src_w guaranteed never to be below zero? And is the result of np.zeros() something useful when one of the output dimensions is zero?

Reply:
Correct, you cannot have a negatively sized array.

>And is the result of np.zeros() something useful when one of the output dimensions is zero?

Certainly more useful than an error and it'll be what you asked for.

Reply:
Ok.